x-pack/solutions/observability/plugins/streams/server/routes/create_server_route.ts

```diff
@@ -6,6 +6,39 @@
  */
 
 import { createServerRouteFactory } from '@kbn/server-route-repository';
+import { CreateServerRouteFactory } from '@kbn/server-route-repository-utils/src/typings';
+import { badRequest, forbidden, internal, notFound } from '@hapi/boom';
+import { errors } from '@elastic/elasticsearch';
 import { StreamsRouteHandlerResources } from './types';
+import { StatusError } from '../lib/streams/errors/status_error';
 
-export const createServerRoute = createServerRouteFactory<StreamsRouteHandlerResources>();
+const createPlainStreamsServerRoute = createServerRouteFactory<StreamsRouteHandlerResources>();
+
+export const createServerRoute: CreateServerRouteFactory<
```

Comment:
Not trying to make things more complicated, but this seems very useful in general - should we move this and `StatusError` into the helper package?

Not a blocker.

Reply:
The SLO plugin did something like this a bit back, I guess we're still waiting for the third use case to generalize over.

Reply:
Hmm I just had a look, the way it's implemented is that it loses all type fidelity